compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -824,6 +824,11 @@ private Result seekMatch(ParserRuleContext currentCtx, int lookahead, int curren
                 case UNARY_OPERATOR:
                     hasMatch = isUnaryOperator(nextToken);
                     break;
+                case IS_KEYWORD:
+                    hasMatch = nextToken.kind == SyntaxKind.IS_KEYWORD;
+                    break;
+                case IS_EXPRESSION:
+                    return seekInIsExpression(currentCtx, lookahead, currentDepth, matchingRulesCount);
```

Comment:
Can't we handle this in a similar way to the binary expression?

Reply:
that means we don't need `IS_EXPRESSION` in `ParserRuleContext[] EXPRESSIONS' array?

Reply:
I was mimicking `ACCESS_EXPRESSION` there btw

Reply:
I see. Ok, let's keep this for now. 
I think we will have to refactor both of these to be aligned with the binary expression. Can you please create an issue?

Reply:
sure.